larray_editor/editor.py

```diff
@@ -272,13 +263,13 @@ def setup_menu_bar(self):
         file_menu.addAction(create_action(self, _('Save Data &As'), triggered=self.save_data_as,
                                           statustip=_('Save all arrays as a session in a file')))
         recent_files_menu = file_menu.addMenu("Open &Recent Data")
-        for action in self.recent_file_actions:
+        for action in self.recent_data_files.actions:
             action.setVisible(False)
             action.triggered.connect(self.open_recent_file)
             recent_files_menu.addAction(action)
-        self.update_recent_file_actions()
+        self.recent_data_files.update_actions()
```

Comment:
IMO, update_actions should be a private method of RecentFileList and should never be called explicitly outside of that class.